airflow/executors/base_executor.py

```diff
@@ -149,7 +155,7 @@ def queue_command(
             self.log.info("Adding to queue: %s", command)
             self.queued_tasks[task_instance.key] = (command, priority, queue, task_instance)
         else:
-            self.log.error("could not queue task %s", task_instance.key)
+            self.task_context_logger.error("could not queue task %s", task_instance.key, ti=task_instance)
```

Comment:
is this truly an exceptional circumstance?  i ask because, we should just be careful about adding too many usages of `task_context_logger` because, in many cases, each call results in uploading a file to s3.  think about what happens if the scheduler starts having to upload files for many TIs in the scheduler loop.

ideally, people use a better logging system like elastic, where, it's at least possible that "shipping logs" just means printing to stdout.  but i don't think that's true for the majority of users.

perhaps this is an other argument for considering sticking it in a queue that is managed separately from the main scheduler loop.  just a thought.

Reply:
here's another thought for you. perhaps instead of using task context logger for every failure, we should add a `reason` field to go along with `state`, so that we can add a small message about what happened / why the task ended up in the state that it did, and then we would not need to use this, unfortunately clunky and not-super-good-for-performance feature.

Reply:
I like the idea and I think it can be valuable for Airflow users, but that'd deserve an AIP (or at least a message de dev list to ask feedbacks). Though, I still do think we need to send logs to task logs in executors (and also other components of Airflow). If the executor is unable to do something about a given task, the user should know about it with all the possible details (stacktrace), so I think we still have this problem we need to solve